New start v2

SkylerPIlot opened the pull request.

Commit:
ready for teaching

-clean of old mistakes

SkylerPIlot committed:
Added Button IDs for better readability

SkylerPIlot committed:
Added commands and variables for the Elevator subsystem.

Added Sets and Toggles for the Brake and Haysqueze commands.

SkylerPIlot committed:
Finished Elevator Subsystem

SkylerPIlot committed:
Made starting changes to Intake Subsystem

Added functions and commands for toggling and setting the actuator in the Intake Subsystem

SkylerPIlot committed:
Almost finished Intake Subsystem

Added command to control intake and its relevant function

SkylerPIlot committed:
Finished lights subsystem

SkylerPIlot committed:
fixed some problems

-copy pasta in defines in header files
-missing requires
-finished pneumatic sensor check
-separated intake and elevatoron into auto and teleop

SkylerPIlot committed:
Cleaned up the Code

Added some needed inclusions and took out all robotbuilder declerations that were uneeded.

Commit:
Merge branch 'feature/olivia/fixes' into newStart-v2

# Conflicts:
#	src/Subsystems/Intake.cpp

SkylerPIlot committed:
Debug

Commit:
Fix for SF

-removed pneumatic sensors
-fixed chassis default command

Commit:
still crashes

Commit:
stopped using robotdrive because it crashed

Commit:
codriver controls added to driver

SkylerPIlot committed:
Fixed up the code

Fixed the logic error and cleaned up a Enum

Commit:
SF tweaks

-tank drive
-one controller controls

Commit:
SF tweaks

-tank drive
-one controller controls (reverted from commit f45f32eae805cad29e78c60c990076d96a3371a6)

SkylerPIlot committed:
Cleaned up RobotBuilder Tags.

Cleaned up all Robotbuilder Autogenerated code and fixed the unused LightsOn command.

duane701 commented on src/OI.cpp:
```diff
@@ -22,59 +22,62 @@
+    coLGreen.reset(new JoystickButton(coDriver.get(), kLGreen_ID));
     coLGreen->WhileHeld(new AutonomousCommand());
-    coLRed.reset(new JoystickButton(coDriver.get(), 12));
+    coLRed.reset(new JoystickButton(coDriver.get(), kLRed_ID));
```
`kLRed`? Does that stand for `kLeftRed`? 

Avoid abbreviations (that aren't totally obvious).

No change required.

duane701 commented on src/OI.cpp:
```diff
@@ -22,59 +22,62 @@
+    coButtonBack.reset(new JoystickButton(coDriver.get(), kButtonBack_ID));
     coButtonBack->WhileHeld(new AutonomousCommand());
-    coButtonRB.reset(new JoystickButton(coDriver.get(), 6));
+    coButtonRB.reset(new JoystickButton(coDriver.get(), kButtonRB_ID));
```
`kButtonRB_ID`? What is the `RB`? Same problem. Avoid abbreviations.

(No change required.)

duane701 commented on src/Subsystems/Elevator.cpp:
```diff
@@ -43,3 +43,5 @@ void Elevator::InitDefaultCommand() {
 // Put methods for controlling this subsystem
 // here. Call these from Commands.
+
+void Elevator::
```
This should have been removed. I expect that a future commit fixes this.

duane701 commented on src/Commands/ElevatorOn.h:
```diff
@@ -0,0 +1,21 @@
+#ifndef ElevatorOn_H
+#define ElevatorOn_H
+
+#include "Commands/Subsystem.h"
```
`#include "Commands/Subsystem.h"` should be `#include "Commands/Command.h"`

duane701 commented on src/Subsystems/Elevator.h:
```diff
@@ -23,21 +23,41 @@ class Elevator: public Subsystem {
+
+	//...........haysqueeze..........
+
+	bool IsHaySqueezeOpen();
```
Should be `bool IsHaySqueezeOpen() const;`

Repeat for all `Is` functions or getters.

duane701 commented on src/Commands/SetHaySqueeze.cpp:
```diff
@@ -16,7 +16,7 @@ void SetHaySqueeze::Initialize()
 void SetHaySqueeze::Execute()
 {
-
+Robot::elevator->SetHaySqueeze(mValue);
```
Indent.

duane701 commented on src/Commands/ToggleHaySqueeze.cpp:
```diff
@@ -0,0 +1,40 @@
+ToggleHaySqueeze::ToggleHaySqueeze()
+{
+	// Use Requires() here to declare subsystem dependencies
+	// eg. Requires(chassis);
```
`Requires()` is required.

duane701 commented on src/Commands/SetActuator.cpp:
```diff
@@ -0,0 +1,39 @@
+: mValue(value)
+{
+	// Use Requires() here to declare subsystem dependencies
+	// eg. Requires(chassis);
```
`Requires()`. I won't repeat this comment any longer. Every command should have a requires.

duane701 commented on src/Commands/SetActuator.h:
```diff
@@ -0,0 +1,19 @@
+#ifndef SetBrake_H
+#define SetBrake_H
+
+#include "Commands/Subsystem.h"
```
Use 

`#include "Commands/Command.h"`

I won't repeat this any longer. Every command header file should use this command.

duane701 commented on src/Subsystems/Intake.h:
```diff
@@ -23,18 +23,25 @@ class Intake: public Subsystem {
-    // END AUTOGENERATED CODE, SOURCE=ROBOTBUILDER ID=CONSTANTS
+	//...........Actuator..........
+
+	bool IsActuatorClosed();
```
Use 

`bool IsActuatorClosed() const;`

I won't comment on these any longer. Find all getters and make them `const`.

duane701 commented on src/Subsystems/Lights.cpp:
```diff
@@ -26,6 +26,9 @@ Lights::Lights() : Subsystem("Lights") {
 
+    heimdallGlow->Set(Relay::kForward);
+    leftUnderGlow->Set(Relay::kForward);
+    rightUnderGlow->Set(Relay::kForward);
```
Should the `Lights` subsystem declare it's own type for `Set`? Or we should use `Relay::kOn` instead of `kFroward.

duane701 commented on src/OI.cpp:
```diff
@@ -43,6 +45,8 @@ OI::OI() {
     coRYellow.reset(new JoystickButton(coDriver.get(), kRYellow_ID));
     coRYellow->WhileHeld(new AutonomousCommand());
+
+    /*
```
Is this code unused? 

Delete unused code.

SkylerPIlot replied:
This code will be used for a example of sendable choosers later.

duane701 reviewed (changes requested):
FTP

Try to have commits "Tell a story."
Work on commits being related to each other. You don't have to check in all of the changed files at once. You can check in some files and not others. Then check the others in later.

Tab/Space. Set the environment to 2 spaces, no tabs.

SkylerPIlot committed:
Consts

Added Consts in their required places

SkylerPIlot committed:
Fixed inclusions for Commands

Fixed Require() error and fixed indent error also

SkylerPIlot committed:
Light

added a enum value for lights subsystem and implemented it.

Diego-Andrade commented on src/Commands/LightsOn.cpp:
```diff
@@ -1,38 +1,39 @@
+: mValue(value)
 {
 	// Use Requires() here to declare subsystem dependencies
 	// eg. Requires(chassis);
```
Requires?

Diego-Andrade commented on src/Commands/LightsOn.h:
```diff
@@ -0,0 +1,21 @@
+	void Interrupted();
+
+private:
+	Lights::LightValue mValue;
```
I believe you want to include Lights.h, to make it clear what your class depends on by quick glance at the includes

Diego-Andrade commented on src/Commands/SetActuator.h:
```diff
@@ -0,0 +1,21 @@
+	void Interrupted();
+
+private:
+	Intake::ActuatorValue mValue;
```
Like before, consider adding Intake.h as an include

Diego-Andrade commented on src/Commands/SetBrake.h:
```diff
@@ -0,0 +1,21 @@
+	void Interrupted();
+
+private:
+	Elevator::BrakeValue mValue;
```
Once more, adding Elevator.h to the includes give you a better idea of the dependencies of your class.

Diego-Andrade commented on src/Commands/SetHaySqueeze.h:
```diff
@@ -0,0 +1,21 @@
+	void Interrupted();
+
+private:
+	Elevator::HaySqueezeValue mValue;
```
By this point you should know where this is going... Consider adding Elevator.h to the includes

Diego-Andrade commented on src/Commands/ToggleActuator.cpp:
```diff
@@ -0,0 +1,42 @@
+// Called repeatedly when this Command is scheduled to run
+void ToggleActuator::Execute()
+{
+	Intake::ActuatorValue value
```
Consider adding Intake.h to includes!!

Diego-Andrade commented on src/Commands/ToggleBrake.cpp:
```diff
@@ -0,0 +1,41 @@
+// Called repeatedly when this Command is scheduled to run
+void ToggleBrake::Execute()
+{
+	Elevator::BrakeValue value
```
Consider adding Elevator.h to includes

Diego-Andrade commented on src/Commands/ToggleHaySqueeze.cpp:
```diff
@@ -0,0 +1,43 @@
+{
+	Requires(Robot::elevator.get());
+	// Use Requires() here to declare subsystem dependencies
+	// eg. Requires(chassis);
```
Requires!

Diego-Andrade commented on src/Commands/ToggleHaySqueeze.cpp:
```diff
@@ -0,0 +1,43 @@
+// Called repeatedly when this Command is scheduled to run
+void ToggleHaySqueeze::Execute()
+{
+	Elevator::HaySqueezeValue value
```
(synonym for consider) adding Elevator.h!

Diego-Andrade commented on src/OI.cpp:
```diff
@@ -1,101 +1,92 @@
-    coLGreen.reset(new JoystickButton(coDriver.get(), 15));
+
+ //TODO find out and replace all old button controls
+	coDriver.reset(new Joystick(1));
```
Magic Number? Meaning if we make a constant for the button ids we might as well make one for the controllers

Diego-Andrade commented on src/OI.cpp:
```diff
@@ -1,101 +1,92 @@
+    */
+
+
     driver.reset(new Joystick(0));
```
Magic Number!

Diego-Andrade commented on src/OI.h:
```diff
@@ -45,16 +34,35 @@ class OI {
+	static const int kRGreen_ID = 4;
+	static const int kRBlack_ID = 3;
+	static const int kRRed_ID = 1;
+	static const int kRYellow_ID = 2;
```
I assume this is not random chaos? Just thinking of ways to organize them to make it easier to find

Diego-Andrade commented on src/Robot.cpp:
```diff
@@ -1,34 +1,19 @@
-    // END AUTOGENERATED CODE, SOURCE=ROBOTBUILDER ID=INITIALIZATION
-
 void Robot::RobotInit() {
 	RobotMap::init();
```
My OCD!!! Not aligned

Diego-Andrade commented on src/Robot.cpp:
```diff
@@ -1,34 +1,19 @@
 	RobotMap::init();
-	// BEGIN AUTOGENERATED CODE, SOURCE=ROBOTBUILDER ID=CONSTRUCTORS
-    elevator.reset(new Elevator());
+	elevator.reset(new Elevator());
```
Another hit to my OCD

Diego-Andrade commented on src/Robot.cpp:
```diff
@@ -37,11 +22,10 @@ void Robot::RobotInit() {
 
-    // END AUTOGENERATED CODE, SOURCE=ROBOTBUILDER ID=AUTONOMOUS
-  }
+	}
```
One to many tabs!

Diego-Andrade commented on src/Robot.h:
```diff
@@ -1,45 +1,28 @@
 public:
 	std::unique_ptr<Command> autonomousCommand;
 	static std::unique_ptr<OI> oi;
 	LiveWindow *lw = LiveWindow::GetInstance();
```
SPACING!

Diego-Andrade commented on src/Robot.h:
```diff
@@ -1,45 +1,28 @@
+
 	virtual void RobotInit();
 	virtual void DisabledInit();
 	virtual void DisabledPeriodic();
```
Tabbing again, although this was auto generated

Diego-Andrade commented on src/RobotMap.cpp:
```diff
@@ -1,99 +1,105 @@
+
+    //..........elevator..........
+
+    elevatorHaySqueeze.reset(new DoubleSolenoid(0, kHaySqueezeF, kHaySqueezeR));
```
I mean this one is more acceptable, the magic number, as we only use one PCM... buuuutt we could still give it a const. (It refers to the PCM id in case you were wondering)

Diego-Andrade commented on src/Subsystems/Chassis.cpp:
```diff
@@ -1,52 +1,35 @@
-    leftRearWheel = RobotMap::chassisleftRearWheel;
-    rightFrontWheel = RobotMap::chassisrightFrontWheel;
-    rightRearWheel = RobotMap::chassisrightRearWheel;
+	leftLightSensor = RobotMap::chassisLeftLightSensor;
```
Spacing

Diego-Andrade commented on src/RobotMap.h:
```diff
@@ -22,27 +11,105 @@
+	static const int kLeftLightID = 1;
+	static const int kRightLightID = 2;
 
 	static void init();
```
This tabbing looks off to me. It is collectively off, which is good at least. That or I'm crazy

Diego-Andrade commented on src/Subsystems/Chassis.cpp:
```diff
@@ -1,52 +1,35 @@
-
-
-    // END AUTOGENERATED CODE, SOURCE=ROBOTBUILDER ID=DEFAULT_COMMAND
+	SetDefaultCommand(new MecanumDrive());
```
spacing

Diego-Andrade commented on src/Subsystems/Chassis.cpp:
```diff
@@ -1,52 +1,35 @@
+	leftRearWheel->Set(-leftSpeed);
+	rightFrontWheel->Set(-rightSpeed);
+	rightRearWheel->Set(-rightSpeed);
+
```
spacing

Diego-Andrade commented on src/Subsystems/Chassis.h:
```diff
@@ -23,26 +12,21 @@ class Chassis: public Subsystem {
-    // END AUTOGENERATED CODE, SOURCE=ROBOTBUILDER ID=DECLARATIONS
 
 	std::shared_ptr<RobotDrive> robotDrive;
 
```
more spacing but from this point forward, I'm not going to address this individually.

Diego-Andrade reviewed (changes requested):
One real issue to be address. The rest are style and suggestions

Diego-Andrade commented:
A comment about my comments...

Maybe @duane701 can better explain, but you normally want to include what you will be using, in the spot where you will using it. There are times where Robot.h is included and not referenced to at all. Of course this is autogenerated and we hardly look at this, but it is something to keep in mind. I believe it shortens build time as well to have the compiler only include what you need. It will work as of now, but like for example, to access enums of certain classs we are "tunneling" through Robot.h then to the class that includes the enum. A bit inefficient. 
My comments point to the bit of code that requires another class ( I might have missed some). So you just want to look at all includes and clean them up a little.

The comments, as @duane701 has said, about spacing are easier to fix. Eclipse has a setting for the style you want to use in your files. You can change it their and it will apply to the whole project and fix the tabbing issues. It's somewhere under File->Properties->C++ General->Formatter. Click on the project first!

SkylerPIlot committed:
Made it easier to determine what subsystem each command depends on.

SkylerPIlot committed:
Fied Requires issue and made code have more readability

Diego-Andrade committed:
Added example formatter file

Used Eclipse to generate the file, which you can load into any project by going to the projects properties and importing a file to the formatter second. 

>File->Properties->C++ General->Formatter

Diego-Andrade committed:
Fixed spacing and gave include examples

Includes are only included where they are needed. There are some excpetions to this rule, but do what is easiest to fix later on down the line.

Diego-Andrade commented:
Quick note, some tabs were left in the code. Meaning that although eclipse see them as 2 spaces, github does not. I double checked, let me know if I missed some however.

duane701 commented on src/Commands/AutonomousCommand.h:
```diff
@@ -1,20 +1,9 @@
-#include "Commands/Subsystem.h"
-#include "../Robot.h"
+#include "Commands/Command.h"
+//#include "../Robot.h"
```
Remove commented code.

duane701 commented on src/Commands/ElevatorOn.h:
```diff
@@ -0,0 +1,21 @@
+#define ElevatorOn_H
+
+#include "Commands/Command.h"
+//#include "../Robot.h"
```
Remove unused code.

duane701 commented on src/Commands/IntakeOn.h:
```diff
@@ -0,0 +1,21 @@
+
+#include "Commands/Command.h"
+//#include "../Robot.h"
+//#include "Subsystems/Intake.h"
```
Remove unused code.

duane701 commented on src/Commands/LightsOn.h:
```diff
@@ -0,0 +1,22 @@
+#define LightsOn_H
+
+#include "Commands/Command.h"
+//#include "../Robot.h"
```
Remove unused code.

duane701 commented on src/Commands/MecanumDrive.h:
```diff
@@ -1,20 +1,10 @@
-#include "../Robot.h"
+#include "Commands/Command.h"
+//#include "../Robot.h"
+//#include "Subsystems/Chassis.h"
```
Remove unused code.

duane701 reviewed (changes requested):
Looks like several places where old includes needs to be removed.

FTP

Diego-Andrade commented:
Correct, @SkylerPIlot They give you an idea of what they replace, and remove them once you see and agree why they were replaced

Commit:
Removed all unneeded include tags that were already commented out.

duane701 commented:
I think there are more changes that could be made, but this is pretty close. 

Smaller is better and easier to review. Keep this in mind moving forward.
